### ISSUE TITLE
Login error

### ISSUE THREAD

**Issue by waigs** [netlify/netlify-cms#2241]
I'm getting the folowing error after login:
`Invariant Violation: Minified React error #31; visit https://reactjs.org/docs/error-decoder.html?invariant=31&args[]=TypeError%3A%20Cannot%20read%20property%20'expires_at'%20of%20undefined&args[]= for the full message or use the non-minified dev environment for full errors and additional helpful warnings.`

This is hapening on `netlify-cms`: **2.6.1**

**Comment by tomrutgers** [comment]
@waigs what was causing the error and what solved it?

**Comment by obedm503** [comment]
Having the exact same problem

**Comment by sbuys** [comment]
Having the same issue...

**Comment by fleadope** [comment]
bump

**Comment by tomrutgers** [comment]
Reopening
Please share your configuration so we can get to the bottom of this

**Comment by obedm503** [comment]
Just to give more details. 

this is only happening in development, on a local server.  the `/admin` login page tries to call `/.netlify/identity/token`. this fails probably because there is no such a route locally. While on netlify, it's provided by netlify itself.

My current theory is that perhaps `/admin` is not supposed to be used in a local development environment. But then, this is not mentioned anywhere (at least I haven't seen it).

**Comment by tomrutgers** [comment]
@obedm503 if you clear your cache and enter the url of your live environment on Netlify (https://www.yoursite.netlify.com/), does that fix the error @obedm503?

**Comment by obedm503** [comment]
@tomrutgers it works perfectly fine in the live environment. it only breaks when running a local server. 

I'm using the gatsby template and running `gatsby serve` command

**Comment by talves** [comment]
@obedm503 what version of `netlify-cms` is getting served up? can you look in the inspector console and report

**Comment by paulkn** [comment]
I am having this issue on my live site once I transferred to my own domain.

**Comment by tomrutgers** [comment]
@obedm503 Sorry for the confusion but I was talking about localhost. If you visit your admin login on localhost, git gateway asks where it can find the live site to make connection to the Netlify Identity settings. If you clear your cache, you get asked for this url again.

**Comment by obedm503** [comment]
@talves  not sure what you mean by the "inspector console", but I have version `2.8.0` in my `node_modules`

**Comment by talves** [comment]
Ok, so 2.8.0 has an error for sure. Can you do a `yarn add netlify-cms@beta` from the terminal and test it out to see if it fixes it. The 2.8.0 has a breaking change for the gatsby plugin.

**Comment by obedm503** [comment]
tried it, and it works!

I tried login in and editing a page, but it edited the files on github instead of the local pages. perhaps that's for a different issue

**Comment by talves** [comment]
@obedm503 Ok. Glad it works now.

You realize it will not write changes to your local file system. See https://github.com/netlify/netlify-cms/issues/329

I have a beta file-system backend that works with Gatsby, but it's waiting on some changes to the plugin before I release it for general use.

**Comment by talves** [comment]
@paulkn can you try the beta release also. I am going to close this for now.

**Comment by talves** [comment]
`netlify-cms@2.9.0` has been released. You should all see these fixed once you bump to the latest version. Please open a new issue for any errors.
